tests/styleguide.spec.ts

```diff
@@ -46,40 +47,51 @@ test.afterEach(async () => {
 });
 
 // Iterate over all sample sections and take a screenshot of each one.
-test('UI regression test - Styleguide sections', async () => {
+test('UI regression test - Styleguide sections', async ({ browser }) => {
   for (let i = 0; i < sampleSectionIds.length; i += 1) {
+    // Can't rely on beforeEach here since we are running tests in a loop inside
+    // single test
+    page = await browser.newPage();
+
     const id = sampleSectionIds[i];
 
     // Isolate the section
     await page.goto(`/ide/styleguide?isolateSection=true#${id}`);
 
-    // Have to reload since we are calling in a loop and only the hash is changing
-    await page.reload();
-
     const sampleSection = page.locator(`#${id}`);
 
     await expect(sampleSection).toHaveScreenshot(
       `${id.replace(/^sample-section-/, '')}.png`
     );
+
+    page.close();
   }
 });
 
-test('Buttons regression test', async () => {
+test('Buttons regression test', async ({ browser }) => {
+  expect(buttonSectionIds.length).toEqual(EXPECTED_BUTTON_SECTION_COUNT);
+
   // Test focus and hover states for each enabled button
   for (let i = 0; i < buttonSectionIds.length; i += 1) {
+    // Can't rely on beforeEach here since we are running tests in a loop inside
+    // single test
+    page = await browser.newPage();
+
     const id = buttonSectionIds[i];
 
     // Isolate the section
-    await page.goto(`/ide/styleguide?isolateSection=true#${id}`);
-
-    // Need to reload since we are calling in a loop and only the hash is changing
-    await page.reload();
+    const sectionUrl = `/ide/styleguide?isolateSection=true#${id}`;
+    await page.goto(sectionUrl);
 
     const section = page.locator(`#${id}`);
+    await section.waitFor();
```

Comment:
This is needed to ensure button tests run

Reply:
you should be awaiting most of these page actions, then this is not needed.

Just `await page.locator`

https://playwright.dev/docs/locators

Reply:
@dsmmcken have you seen any examples of awaiting the `page.locator()` call? I've only seen ones awaiting methods off of that, and it doesn't seem to return a promise

Reply:
For reference, these are the docs I followed for this particular case:
https://playwright.dev/docs/api/class-locator#locator-wait-for

Reply:
```
 const grid = await page.locator('.iris-grid-panel .iris-grid');
 ```
hmm, I did this in another test and it is working? 🤷  Maybe that isn't actually awaiting the locator, I don't know. But it runs.